src/domain/course/__tests__/CoursePageContainer.test.tsx

```diff
@@ -243,3 +281,29 @@ it('should link to courses search when clicking tags', async () => {
     search: '?text=Avouinti',
   });
 });
+
+it('should contain event hero with super event date', async () => {
+  advanceTo('2020-06-23');
+
+  renderWithRoute(<CoursePageContainer />, {
+    mocks: superEventMocks,
+    routes,
+    path: ROUTES.COURSE,
+  });
+
+  await waitFor(() => {
+    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+  });
+
+  const superDateStr = getDateRangeStr({
+    start: superEventResponse.data.courseDetails.startTime,
+    end: superEventResponse.data.courseDetails.endTime,
+    locale: 'fi',
+    includeTime: true,
+    timeAbbreviation: translations.commons.timeAbbreviation,
+  });
+
+  expect(
+    screen.getAllByText((_content, el) => el.textContent === superDateStr)
+  ).toHaveLength(1);
```

Comment:
This would work also:
``` 
 expect(
    screen.getByText((_content, el) => el.textContent === superDateStr)
  ).toBeInTheDocument();
```

Also adding data-testid and querying with that might be helpful.

Reply:
toBeInTheDocument() does not work, because returned type is array with one element.

Reply:
fixed